server/api.js

```diff
@@ -699,31 +699,53 @@ router.post("/bid", async (req, res) => {
 			const checkBidQuery = `
 				SELECT * FROM bid WHERE tender_id = $1 AND bidder_id = $2
 			`;
+
 			const checkBidValues = [tenderId, bidderId];
 			const existingBid = await client.query(checkBidQuery, checkBidValues);
 
-			if (existingBid.rows.length > 0) {
-				await client.query("ROLLBACK");
-				return res.status(400).json({ code: "DUPLICATE_ENTRY" });
-			}
+			let bidResult;
 
-			const bidQuery = `
+			if (existingBid.rows.length > 0) {
```

Comment:
If we are going to allow bidders to submit a new bid after withdrawing, then we should just insert a new bid, not update the existing one.